components/sup/src/manager/service/mod.rs

```diff
@@ -92,7 +93,7 @@ pub struct Service {
     last_health_check: Instant,
     #[serde(skip_serializing)]
     manager_fs_cfg: Arc<manager::FsCfg>,
-    supervisor: Supervisor,
+    pub supervisor: Supervisor,
```

Comment:
We should keep this private and expose some public member functions on the `Service` struct to expose the API that you need. The `Supervisor` module should be considered internal